src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -586,6 +588,116 @@ def query_for_quota_alert(self):
             for dataset in self.query()
         ]
 
+    def available_zvols_fast(self, options=None):
```

Comment:
How about perhaps adding an `accepts` decorator which can do some schema validation to ensure devs don't make mistakes when calling this internally ?